clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -90,7 +90,7 @@ public FileOutStream(long fileId, OutStreamOptions options) throws IOException {
       mUfsPath = fileInfo.getUfsPath();
       String fileName = PathUtils.temporaryFileName(fileId, mNonce, mUfsPath);
       UnderFileSystem ufs = UnderFileSystem.get(fileName, ClientContext.getConf());
-      String parentPath = (new TachyonURI(mUfsPath)).getParent().getPath();
+      String parentPath = (new TachyonURI(mUfsPath)).getParent().toString();
       if (!ufs.exists(parentPath) && !ufs.mkdirs(parentPath, true)) {
```

Comment:
Why does the original code work with HDFS, but not with S3? Shouldn't they have the similar issue with the scheme preservation?